rusqlite_migration/src/lib.rs

```diff
@@ -161,6 +154,16 @@ pub struct M<'u> {
     foreign_key_check: bool,
 }
 
+impl<'u> PartialEq for M<'u> {
+    fn eq(&self, other: &Self) -> bool {
+        self.up == other.up
+            && self.down == other.down
+            && self.foreign_key_check == other.foreign_key_check
```

Comment:
Let’s discuss this in the issue: https://github.com/cljoly/rusqlite_migration/issues/45#issuecomment-1471024204

Reply:
(I’m interested in why you went for your first solution in the end :))

Reply:
I had no strong feelings one way or the other, implemented this as a quick and dirty solution for the time being as I waited for your decision. I'll change it according to what you wrote ;).

Reply:
Changed!

Reply:
Great, thanks!